tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Verify/VerifySetupTool.cs

```diff
@@ -0,0 +1,338 @@
+            };
+        }
+
+        logger.LogInformation("Command {Command} succeeded. Output: {Output}", string.Join(' ', command), trimmed);
+
+        return new DefaultCommandResponse
+        {
+            Message = $"Command {string.Join(' ', command)} succeeded. Output: {trimmed}"
+        };
+    }
+
+    private async Task<List<SetupRequirements.Requirement>> GetRequirements(HashSet<SdkLanguage> languages, string packagePath, string venvPath, CancellationToken ct)
+    {
+        // Check core requirements before language-specific requirements
+        var reqsToCheck = await GetCoreRequirements(ct);
+
+        // Per-language requirements
+        var reqGetters = null as List<IEnvRequirementsCheck?>;
+        if (languages == null || languages.Count == 0)
+        {
+            // Detect language if none given
+            reqGetters = new List<IEnvRequirementsCheck?> { await envRequirementsCheck.Resolve(packagePath) };
+
+            if (reqGetters == null || reqGetters.Count == 0)
+            {
+                logger.LogWarning("Could not resolve requirements checker for the specified languages from path {PackagePath}. Checking only core requirements. Please provide languages explicitly to check language requirements.", packagePath);
+                return reqsToCheck;
+            }
+        } else
+        {
+            reqGetters = envRequirementsCheck.Resolve(languages);
+        }
+
+        if (reqGetters == null)
+        {
+            throw new Exception("Could not resolve requirements checker for the specified languages.");
+        }
+
+        foreach (var getter in reqGetters)
+        {
+            if (getter == null)
+            {
+                logger.LogError("Could not resolve requirements checker for one of the specified languages.");
+                continue;
+            }
+
+            if (getter is PythonRequirementsCheck pythonReqCheck && !string.IsNullOrEmpty(venvPath))
+            {
+                // If checking Python and venv path provided, use it
+                reqsToCheck.AddRange(await pythonReqCheck.GetRequirements(packagePath, venvPath, ct));
+                continue;
+            } 
+            
+            reqsToCheck.AddRange(await getter.GetRequirements(packagePath, ct));
+            
+        }
+
+        return reqsToCheck ?? new List<SetupRequirements.Requirement>();
+    }
+
+    private async Task<List<SetupRequirements.Requirement>> GetCoreRequirements(CancellationToken ct)
+    {
+        var assembly = Assembly.GetExecutingAssembly();
+        var names = assembly.GetManifestResourceNames();
+        using var stream = assembly.GetManifestResourceStream("Azure.Sdk.Tools.Cli.Configuration.RequirementsV1.json");
+        using var reader = new StreamReader(stream);
+        var requirementsJson = await reader.ReadToEndAsync(ct);
+
+        var setupRequirements = JsonSerializer.Deserialize<SetupRequirements>(requirementsJson);
```

Comment:
I feel like the config file shouldn't need to be deserialized again for every category of requirements. I think the requirements should be parsed once and then passed to the language implementations